### PR TITLE
deps: Update to streams3 API

### PR DESCRIPTION
**Pull request by legobeat**
#### Changed
- **BREAKING**: Consolidate on streams API v3. All streams are now using and expecting compatibility with `readable-stream@3.x`.
  - https://www.npmjs.com/package/readable-stream#version-3xx
- Update `extension-port-stream` from `^2.1.1` to `^3.0.0`
  - [CHANGELOG](https://github.com/MetaMask/extension-port-stream/blob/main/CHANGELOG.md) 
- Update `json-rpc-middleware-stream` from `^4.2.3` to `^5.0.1`
  - [CHANGELOG](https://github.com/MetaMask/json-rpc-middleware-stream/blob/main/CHANGELOG.md) 
- Update `@metamask/object-multiplex` from `^1.3.0` to `^2.0.0`
  - [CHANGELOG](https://github.com/MetaMask/object-multiplex/blob/main/CHANGELOG.md) 
- Add direct dependency on `readable-stream@^3.6.2`

The package is previously mixing usage of `readable-stream@2`, `readable-stream@3` and Node.js native streams (the API and behavior of which differ depending on Node.js version). This consolidates all internal streams on portable `readable-stream`.

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore extension-port-stream@3.0.0

File: .eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,13 @@ module.exports = {
       },
     },
 
+    {
+      files: ['*.d.ts'],
+      rules: {
+        'import/unambiguous': 'off',
```
Comment (jiexi) [inline .eslintrc.js:22]: why is this needed?

**Comment by jiexi** [comment]
I mean, the tests pass. We probably want to test this locally in extension and see what happens. I imagine it'll just work though

File: .eslintrc.js (hunk 1 above)
Comment (legobeat) [reply .eslintrc.js:22]: Otherwise we get (for the only `*.d.ts` file in the project):

```
src/readable-stream.d.ts
  1:1  error  This module could be parsed as a valid script  import/unambiguous
```

**Comment by jiexi** [comment]
Tested locally. Seems to work fine

**Review by jiexi — APPROVED** [review]
